firebase_admin/db.py

```diff
@@ -199,6 +235,35 @@ def delete(self):
         """
         self._client.request_oneway('delete', self._add_suffix())
 
+    def transaction(self, transaction_update):
+        """Write to database using a transaction.
+
+        Args:
+            transaction_update: function that takes in current database data as a parameter.
+
+        Returns:
+            bool: True if transaction is successful, otherwise False.
+
+        Raises:
+            ValueError: If transaction_update is not a function.
+
+        """
+        if not callable(transaction_update):
+            raise ValueError('transaction_update must be a function.')
+
+        tries = 0
+        etag, data = self._get_with_etag()
+        val = transaction_update(data)
```

Comment:
We should not move `_get_with_etag()` into the loop as it makes a REST call. Perhaps:

```
etag, data = self._get_with_etag()
while tries < _TRANSACTION_MAX_ENTRIES:
    new_data = transaction_update(data)
    success, etag, data = self._update_with_etag(new_data, etag)
    if success:
        return True
    tries += 1
```